cmd/juju/application/config.go

```diff
@@ -343,9 +343,13 @@ func (c *configCommand) getConfig(client configCommandAPI, ctx *cmd.Context) err
 		}
 		v, ok := info["value"]
 		if !ok || v == nil {
-			v = ""
+			// Note that unlike the output with a non-nil and non-empty value below, this
+			// output should never have a newline.
+			fmt.Fprint(ctx.Stdout, "")
```

Comment:
um... saying print nothing does nothing and you are missing the return :)

Perhaps just a return statement?